File: TypeScript/src/repository/userRepository.ts
import { UserModel, User } from "../models/userModel";

export interface UserRepository {
    create(username:string, password:string, name:string): Promise<void>;
    findByUsernameAndPassword(username:string, password:string): Promise<User | null>;
}


export class UserRepositoryImpl implements UserRepository {
    async create(username:string, password:string, name:string): Promise<void> {
        await UserModel.create(username, password, name);
    }

    async findByUsernameAndPassword(username: string, password: string): Promise<User | null> {
        const user = await UserModel.findByUsernameAndPassword(username, password);
        return user;
    }
}

File: C#/Repositories/UserRepository.cs
using System;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using Microsoft.EntityFrameworkCore.Query;
using TaskManagerApp.Database;
using TaskManagerApp.Models;

namespace TaskManagerApp.Repositories
{
    public class UserRepository
    {
        public static bool UserLogin(string username, string password, AppDbContext? context)
        {
            bool result = Models.UserModel.UserLogin(username, password, context);
            return result;
        }

        public static bool UsernameExists(string username, AppDbContext context)
        {
            bool result = Models.UserModel.UserExistance(username, context);
            return result;
        }

        public static bool NewUserRegister(string username, string password, string name, AppDbContext context)
        {
            bool result = Models.UserModel.Register(username, password, name, context);
            return result;
        }
    }
}